server/controllers/profile.js

```diff
@@ -106,3 +114,25 @@ exports.findSitters = asyncHandler(async (req, res) => {
     return res.status(500).json({ message: error });
   }
 });
+
+//@route Get /profile/location/:search
+//return list of profiles who match users search 
+exports.findSittersByLocation = asyncHandler(async (req, res) => {

```

Comment:
I noticed that we have a controller to fetch all sitters (`findSitters`), you can use the same controller to fetch all profiles/by location/ by date.

You can refactor the route to use query parameters:
Eg: `fetch(`/profiles?location=${location}&startDate=${startDate}&endDate=${endDate}`

If location and dates are undefined you would return all the profiles, if they're specified then you would filter data respectively and send a response.

Reference: https://coderrocketfuel.com/article/handle-get-request-query-string-parameters-in-express-js